turbopack/crates/turbopack-resolve/src/node_native_binding.rs

```diff
@@ -132,23 +132,24 @@ pub async fn resolve_node_pre_gyp_files(
                         )
                         .into();
 
-                    for (key, entry) in config_file_dir
+                    if let DirectoryContent::Entries(entries) = &*config_file_dir
                         .join(native_binding_path.clone())
-                        .read_glob(
-                            Glob::new(format!("*.{}", compile_target.dylib_ext()).into()),
-                            false,
-                        )
+                        .read_dir()
                         .await?
-                        .results
-                        .iter()
                     {
-                        if let &DirectoryEntry::File(dylib) | &DirectoryEntry::Symlink(dylib) =
-                            entry
+                        let extension = format!("*.{}", compile_target.dylib_ext());
+                        for (key, entry) in entries
+                            .iter()
+                            .filter(|(k, _)| k.as_str().ends_with(&extension))
```

Comment:
`.ends_with("*.dylib")` sounds wrong? The glob star shouldn't be in there.

Also you shouldn't need `as_str()` because it automatically derefs into a str anyway.

Reply:
And `dylib_ext()` is only used here, so might as well change that function to include the dot before the extension, then you don't need the `format!`

Reply:
🤦‍♂️ Done!